common/auth.go

```diff
@@ -15,6 +15,41 @@
 
 package common
 
-type AuthBackend interface {
-	
-}
\ No newline at end of file
+type Role interface {
+	Name() string
+}
+
+type role struct {
+	name string
+}
+
+// Represents the type of credential (e.g., certificate,
+// username-password, etc.
+type CredentialType string
+
+const (
+	CredentialUsernamePassword = "userPass"
+	CredentialNone             = "none"
+)
+
+// Container for various credentials. Currently containing Username/Password
+// but keys, certificates, etc. can be used in the future.
+type Credential struct {
+	Type     CredentialType
+	Username string
+	Password string
+}
+
+// MakeCredentialFromCliArgs takes all possible CLI
+// arguments that can be provided and constructs appropriate
+// Credential structure. This is just keeping in one
+// method a common functionality that will be in every
+// command.
+func MakeCredentialFromCliArgs(username string, password string) Credential {
```

Comment:
This function doesn't extract anything from the command line, so should this not be called: "MakeUsernamePassCredentials" or something like this? It very specifically takes 'username' and 'password' as parameters, but does also not touch the command line at all...